cmds/core/lockmsrs/lockmsrs.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2012-2020 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// lock-msrs locks a bunch of intel MSRs, all specified in the Intel
+// Software developer's manual. This seems like a good set of bits to
+// lock down when booting through NERF/LINUXBOOT to some other OS.
+// When locked, these MSRs generally prevent further modifications until
+// reset.
+package main
+
+import (
+	"log"
+
+	"github.com/intel-go/cpuid"
+	"github.com/u-root/u-root/pkg/msr"
+)
+
+const (
+	msrIA32FeatureControl  msr.MSR = 0x3A  // MSR_IA32_FEATURE_CONTROL
+	msrPkgCstConfigControl msr.MSR = 0xE2  // MSR_PKG_CST_CONFIG_CONTROL
+	msrFeatureConfig       msr.MSR = 0x13C // MSR_FEATURE_CONFIG
+	msrDramPowerLimit      msr.MSR = 0x618 // MSR_DRAM_POWER_LIMIT
+	msrConfigTDPControl    msr.MSR = 0x64B // MSR_CONFIG_TDP_CONTROL
+	ia32DebugInterface     msr.MSR = 0xC80 // IA32_DEBUG_INTERFACE
+)
+
+var msrList = []struct {
+	msrAdd    msr.MSR
+	clearMask uint64
+	setMask   uint64
+}{
+	{
+		// Architectural MSR. All systems.
+		// Enables features like VMX.
+		msrAdd:  msrIA32FeatureControl,
+		setMask: 1 << 0, // Locks this register.
+	},
+	{
+		// Silvermont, Airmont, Nehalem...
+		// Controls Processor C States.
+		msrAdd:  msrPkgCstConfigControl,
+		setMask: 1 << 15, // Locks this register.
+	},
+	{
+		// Westmere onwards.
+		// Note that this turns on AES instructions, however
+		// 3 will turn off AES until reset.
+		msrAdd:  msrFeatureConfig,
+		setMask: 1 << 0,
+	},
+	{
+		// Goldmont, SandyBridge
+		// Controls DRAM power limits. See Intel SDM
+		msrAdd:  msrDramPowerLimit,
+		setMask: 1 << 31, // Locks this register.
+	},
+	{
+		// IvyBridge Onwards.
+		// Not much information in the SDM, seems to control power limits
+		msrAdd:  msrConfigTDPControl,
+		setMask: 1 << 31, // Locks this register.
+	},
+	{
+		// Architectural MSR. All systems.
+		// This is the actual spelling of the MSR in the manual.
+		// Controls availability of silicon debug interfaces
+		msrAdd:  ia32DebugInterface,
+		setMask: 1 << 30, // Locks this register.
+	},
+}
+
+func main() {
+	if cpuid.VendorIdentificatorString != "GenuineIntel" {
+		log.Fatalf("Only Intel CPUs supported, expected GenuineIntel, got %v", cpuid.VendorIdentificatorString)
+	}
+	var errs []error
+
+	cpus := msr.AllCPUs
+	for _, m := range msrList {
+		errs = m.msrAdd.MaskBits(cpus, m.clearMask, m.setMask)
+		for i, e := range errs {
```

Comment:
Would it make sense to have a verbose mode which just prints all the registers and the values they were set to?